scipy/linalg/tests/test_lapack.py

```diff
@@ -3439,3 +3439,24 @@ def test_sy_hetrs(mtype, dtype, lower):
     assert info == 0
     eps = np.finfo(dtype).eps
     assert_allclose(A@x, b, atol=100*n*eps)
+
+
+@pytest.mark.parametrize('norm', list('Mm1OoIiFfEe'))
+@pytest.mark.parametrize('uplo, m, n', [('U', 5, 10), ('U', 10, 10),
+                                        ('L', 10, 5), ('L', 10, 10)])
+@pytest.mark.parametrize('diag', ['N', 'U'])
+@pytest.mark.parametrize('dtype', DTYPES)
+def test_lantr(norm, uplo, m, n, diag, dtype):
+    rng = np.random.default_rng(98426598246982456)
+    A = rng.random(size=(m, n)).astype(dtype)
+    lantr, lange = get_lapack_funcs(('lantr', 'lange'), (A,))
+    res = lantr(norm, A, uplo=uplo, diag=diag)
+
+    # now modify the matrix according to assumptions made by `lantr`
+    A = np.triu(A) if uplo == 'U' else np.tril(A)
+    if diag == 'U':
+        i = np.arange(min(m, n))
+        A[i, i] = 1
+    ref = lange(norm, A)
+
+    assert_allclose(res, ref)
```

Comment:
Huh I'm not sure what was going wrong locally before. Maybe a file was being held and it didn't rebuild or something. 

Anyway, if this looks good, please feel free to commit with `[skip ci]`.
```suggestion
    assert_allclose(res, ref, rtol=1e-6)
```
Or if you'd prefer, we can make the tolerance depend on the dtype. (One job is failing with single precision because the relative error is a hair over `1e-7`.

Reply:
Probably we need the dtype based precision here indeed. Otherwise other architectures might start getting a failures if we keep it just at the crossover. And the rest looks just fine

Reply:
I only asked in case you were concerned with the `1e-6` being so loose for `float64`. I am not concerned with `1e-6` being too tight for `float32` because it's 10x the worst error we're seeing. If you are concerned about that, I'll need a recommendation.

Reply:
I think we don't need too tight bounds anyways since it's a norm problem. It would be either very wrong or just numerical noise difference so I don't mind loose bounds.